models/profiles.yaml

```diff
@@ -452,7 +460,12 @@ var_groups:
             enable_status: must_have
       - entity_var:
           name: session_length
-          select: TIMESTAMPDIFF(SECOND, {{session.Var("session_start_time")}}, {{session.Var("session_end_time")}})
+          select: |
+            {% if warehouse.DatabaseType() == 'bigquery' %}
+              TIMESTAMP_DIFF(CAST({{session.Var("session_end_time")}} AS TIMESTAMP), CAST({{session.Var("session_start_time")}} AS TIMESTAMP), SECOND)
+            {% else %}
+              DATEDIFF(SECOND, CAST({{session.Var("session_start_time")}} AS TIMESTAMP), CAST({{session.Var("session_end_time")}} AS TIMESTAMP))
```

Comment:
With this change, now `DATEDIFF` will be used for Snowflake instead of `TIMESTAMP_DIFF`. Does that mean either syntax is fine for SF ?

Reply:
yeah both are exactly the same, timestamp_diff is actually an alternative for datediff. using datediff saves us a couple of if else statements as it is also the syntax supported by other warehouses.